storage/innobase/xtrabackup/test/t/undo_tablespaces.sh

```diff
@@ -104,3 +102,84 @@ then
     vlog "Checksums do not match"
     exit -1
 fi
+
+#start new test to see if undo_tablespace are moved in datadir if undo_tablespace is not specified
+
+stop_server
+
+rm -rf $MYSQLD_DATADIR/*
+rm -rf $undo_directory_ext/*
+rm -rf $topdir/*
+
+mkdir -p $undo_directory_ext
+
+MYSQLD_EXTRA_MY_CNF_OPTS="
+innodb_file_per_table=1
+innodb_undo_tablespaces=4
+innodb_directories=$undo_directory_ext
+"
+
+start_server
+# create some undo tablespaces
+mysql -e "CREATE UNDO TABLESPACE undo1 ADD DATAFILE '$undo_directory_ext/undo1.ibu'"
+mysql -e "CREATE UNDO TABLESPACE undo2 ADD DATAFILE '$undo_directory_ext/undo2.ibu'"
+mysql -e "CREATE UNDO TABLESPACE undo3 ADD DATAFILE 'undo3.ibu'"
+mysql -e "CREATE UNDO TABLESPACE undo4 ADD DATAFILE 'undo4.ibu'"
+
+mysql -e "ALTER UNDO TABLESPACE innodb_undo_001 SET INACTIVE"
+mysql -e "ALTER UNDO TABLESPACE innodb_undo_002 SET INACTIVE"
+
+load_sakila
+
+checksum1=`checksum_table sakila payment`
+test -n "$checksum1" || die "Failed to checksum table sakila.payment"
+
+ls -al $undo_directory_ext
+# Start a transaction, modify some data and keep it uncommitted for the backup
+# stage. InnoDB avoids using the rollback segment in the system tablespace, if
+# separate undo tablespaces are used, so the test would fail if we did not
+# handle separate undo tablespaces correctly.
+start_uncomitted_transaction &
+job_master=$!
+
+xtrabackup --backup --target-dir=$topdir/backup
+kill -SIGKILL $job_master
+stop_server
+
+rm -rf $MYSQLD_DATADIR/*
+rm -rf $undo_directory_ext/*
```

Comment:
You shouldn't remove the original undo datadir location. This will act as 'running server'

Reply:
okay , I am removing then after the copy back.